sycl/unittests/Extensions/CommandGraph.cpp

```diff
@@ -1933,6 +1935,115 @@ TEST_F(CommandGraphTest, GraphPartitionsMerging) {
   ASSERT_FALSE(PartitionsList[4]->isHostTask());
 }
 
+TEST_F(CommandGraphTest, GetNodeQueries) {
```

Comment:
This might be problematic for `get_root_nodes()` since internally we use a `std::set` for these which will not maintain the correct order of addition. Given that root nodes may be removed from this set at a later stage if they have dependencies added to them, perhaps it is better to just remove this wording from that function.

We could potentially change the way we store the root nodes to maintain ordering, but I'm not sure it actually provides much value to warrant that effort.

Reply:
Good point for root nodes, lets remove the spec wording around for that entry-point.

Reply:
Removed 👍